evalml/objectives/utils.py

```diff
@@ -214,3 +229,158 @@ def get_core_objectives(problem_type):
+                )
+            exclude_objectives.append(ex_obj.name)
+
+    default_objectives.update(set(include_objectives))
+    return default_objectives - set(exclude_objectives)
+
+
+def normalize_objectives(objectives_to_normalize, max_objectives, min_objectives):
+    """Converts objectives from a [0, inf) scale to [0, 1] given a max and min for each objective.
+
+    Args:
+        objectives_to_normalize (dict[str,float]): A dictionary mapping objectives to values
+        max_objectives (dict[str,float]): The mapping of objectives to the maximum values for normalization
+        min_objectives (dict[str,float]): The mapping of objectives to the minimum values for normalization
+
+    Returns:
+        A dictionary mapping objective names to their new normalized values
+    """
+    normalized = {}
+    for objective_name, val in objectives_to_normalize.items():
+        objective_obj = get_objective(objective_name)
+        # Only normalize objectives that are not bounded like percentages
+        # R2 also does not get normalized as it's essentially bounded like a percentage,
+        # and we want to penalize aggressively when R2 is negative
+        if objective_obj.is_bounded_like_percentage or objective_obj.name == "R2":
+            normalized[objective_name] = val
+            continue
+        max_val, min_val = (
+            max_objectives[objective_name],
+            min_objectives[objective_name],
+        )
+        if max_val == min_val:
+            normal = 1
+        else:
+            normal = (val - min_val) / (max_val - min_val)
+            if not objective_obj.greater_is_better:
+                normal = 1 - normal
+        normalized[objective_name] = normal
+    return normalized
+
+
+def recommendation_score(
+    objectives,
+    prioritized_objective=None,
+    prioritized_weight=0.5,
+):
+    """Computes a recommendation score for a model given scores for a group of objectives.
+
+    Args:
+        objectives (dict[str,float]): A dictionary mapping objectives to their values
+        prioritized_objective (str): An optional name of a priority objective that should be given heavier weight
+            than the other objectives contributing to the score. Defaults to None, where all objectives are
+            weighted equally.
+        prioritized_weight (float): The weight to attribute to the prioritized objective, if it exists.
+            Defaults to 0.5.
+
+    Returns:
+        A value between 0 and 100 representing how strongly we recommend a pipeline given a set of evaluated objectives
+
+    Raises:
+        ValueError: If the objective to prioritize is not in the known objectives, or if the priority weight is not
+            a float between 0 and 1.
+    """
+    objectives = objectives.copy()  # Prevent mutation issues
+    priority_weight = 0
+    default_weight = 1 / len(objectives)
+    if prioritized_objective is not None:
+        if prioritized_objective not in objectives:
```

Comment:
`prioritized_objective` allows us to give more (or I guess less) weight to a single objective.  Is it worth designing this function such that it accepts instead an `objectives_weighting (dict[str,float]): Mapping of objectives to their relative weighting to compose the recommendation score` ?

I was kind of thinking that there would existing presets that would function as data scientist selected "blends" of metrics based on what our user was looking for.  This would require weighting of multiple metrics at the same time.

Reply:
That was my original design - amended after discussion [here](https://alteryx.atlassian.net/wiki/spaces/PS/pages/1710916950/Model+Recommendation+Score?focusedCommentId=1720058958)